File: src/main/java/fr/aven/bot/modules/jda/JDAManager.java
package fr.aven.bot.modules.jda;

import com.sedmelluq.discord.lavaplayer.jdaudp.NativeAudioSendFactory;
import fr.aven.bot.Main;
import fr.aven.bot.modules.tickets.TicketsEvent;
import net.dv8tion.jda.api.requests.GatewayIntent;
import net.dv8tion.jda.api.sharding.DefaultShardManagerBuilder;
import net.dv8tion.jda.api.sharding.ShardManager;
import net.dv8tion.jda.api.utils.ChunkingFilter;

public class JDAManager
{
    private static ShardManager shardManager = createShardManager();

    private static ShardManager createShardManager()
    {
        try
        {
            System.out.println(Main.getConfiguration().getString("token", "Insert your token here."));
            return DefaultShardManagerBuilder.createDefault(Main.getConfiguration().getString("token", "Insert your token here."))
                    //.setMemberCachePolicy(MemberCachePolicy.ALL)
                    .setChunkingFilter(ChunkingFilter.ALL)
                    .enableIntents(GatewayIntent.DIRECT_MESSAGE_REACTIONS,
                            GatewayIntent.DIRECT_MESSAGE_TYPING,
                            GatewayIntent.DIRECT_MESSAGES,
                            GatewayIntent.GUILD_BANS,
                            GatewayIntent.GUILD_EMOJIS,
                            GatewayIntent.GUILD_INVITES,
                            GatewayIntent.GUILD_MEMBERS,
                            GatewayIntent.GUILD_MESSAGE_REACTIONS,
                            GatewayIntent.GUILD_MESSAGE_TYPING,
                            GatewayIntent.GUILD_MESSAGES,
                            GatewayIntent.GUILD_VOICE_STATES)
                    .setAudioSendFactory(new NativeAudioSendFactory())
                    .addEventListeners(new TicketsEvent())
                    .build();
        } catch (Exception e)
        {
            Main.getConfiguration().save();
            Main.LOGGER.error("Token erroné.");
        }

        return null;
    }

    public static ShardManager getShardManager() { return shardManager; }
}

File: src/main/java/fr/aven/bot/modules/jda/JDAManager.kt
package fr.aven.bot.modules.jda

import com.sedmelluq.discord.lavaplayer.jdaudp.NativeAudioSendFactory
import fr.aven.bot.Main
import fr.aven.bot.modules.jda.events.*
import fr.aven.bot.modules.tickets.TicketsEvent
import net.dv8tion.jda.api.requests.GatewayIntent
import net.dv8tion.jda.api.sharding.DefaultShardManagerBuilder
import net.dv8tion.jda.api.sharding.ShardManager
import net.dv8tion.jda.api.utils.ChunkingFilter
import net.dv8tion.jda.api.utils.MemberCachePolicy

object JDAManager
{
    @JvmStatic
    var manager: ShardManager = this.buildManager().build()

    private fun buildManager(): DefaultShardManagerBuilder {
        return DefaultShardManagerBuilder
            .createDefault(Main.getConfiguration().getString("token"))
            .enableIntents(GatewayIntent.GUILD_MEMBERS)
            .setAutoReconnect(true)
            .setMemberCachePolicy(MemberCachePolicy.ALL)
            .setChunkingFilter(ChunkingFilter.ALL)
            .setAudioSendFactory(NativeAudioSendFactory())
            .addEventListeners(TicketsEvent(), BingoListener(), AFKListener(), ReactRoleListener(), GuildActivities(), InteractionListener())
    }

    @JvmStatic
    fun getShardManager(): ShardManager { return manager}

}